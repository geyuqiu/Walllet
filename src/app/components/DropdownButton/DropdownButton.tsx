import {Icon} from "app/components/Icon/Icon";
import React from "react";

import {Circle} from '../Circle/Circle';

type DropdownButtonProps = {
	isOpen: boolean;
	label: React.ReactNode
}

export const DropdownButton = ({isOpen, label}: DropdownButtonProps) => {
	return (
		<button className="flex justify-between items-center h-11 w-full">
			<div className="flex items-center">
				<div className="hidden sm:block">
					<Circle className="rounded-l-3xl border-r bg-black-light" size="lg">
						<Icon name="Address" width={15} height={15} fill='#000000' stroke='#FBC457' className="ml-2"
						      data-testid="dropdown__left__icon"/>
					</Circle>
				</div>
				{label}
			</div>
			<Icon width={15} height={15} className="mx-5" data-testid="dropdown__right_icon"
			      name={isOpen ? "ArrowUp" : "ChevronDown"}
			/>
		</button>
	);
};

DropdownButton.defaultProps = {
	isOpen: false,
};
