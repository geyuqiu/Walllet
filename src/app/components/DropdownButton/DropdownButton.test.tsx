import {render} from "@testing-library/react";
import React from "react";

import {DropdownButton} from "./DropdownButton";

describe("WalletCard", () => {
	it("should match snapshot", () => {
		const { container } = render(<DropdownButton/>);

		expect(container).toMatchSnapshot();
	});

});
