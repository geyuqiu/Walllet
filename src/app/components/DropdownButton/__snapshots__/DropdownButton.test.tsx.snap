// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletCard should match snapshot 1`] = `
<div>
  <button
    class="flex justify-between items-center h-11 w-full"
  >
    <div
      class="flex items-center"
    >
      <div
        class="hidden sm:block"
      >
        <div
          class="Circle__CircleWrapper-sc-1hq8kv9-0 cqSuMy rounded-l-3xl border-r bg-black-light ring-theme-background"
        >
          <div
            class="sc-bdfBwQ dfWOLJ ml-2"
            data-testid="dropdown__left__icon"
            height="15"
            width="15"
          >
            <svg
              fill="#000000"
              stroke="#FBC457"
            >
              address.svg
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-bdfBwQ dfWOLJ mx-5"
      data-testid="dropdown__right_icon"
      height="15"
      width="15"
    >
      <svg>
        chevron-down.svg
      </svg>
    </div>
  </button>
</div>
`;
